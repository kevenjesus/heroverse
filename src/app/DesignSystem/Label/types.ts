import { colorsType } from "@/app/styles/types";
import { ReactNode } from "react";


export interface LabelItemProps {
    variant: colorsType
}

export interface LabelProps extends LabelItemProps {
    children: ReactNode
}

