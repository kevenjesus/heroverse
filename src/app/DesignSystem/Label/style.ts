import styled from 'styled-components'
import { LabelItemProps } from './types'
import { colors } from '@/app/styles/guide'

export const Wrapper = styled.div`
    height: 15px;
    display: flex;
    align-items: center;
    gap: 10px;
`

export const LabelBox = styled.div<LabelItemProps>`
    width: 12px;
    height: 12px;
    border-radius: 12px;
    background: ${({variant}) => colors[variant]};
`