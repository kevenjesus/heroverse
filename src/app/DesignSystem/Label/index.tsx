import Typographic from '../Typographic'
import * as S from './style'
import { LabelProps } from './types'

export default function Label({children, variant}: LabelProps) {
    return (
        <S.Wrapper>
            <S.LabelBox variant={variant} />
            <Typographic>{children}</Typographic>
        </S.Wrapper>
    )
}