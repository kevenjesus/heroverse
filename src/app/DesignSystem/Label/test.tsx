import React from 'react';
import { render } from '@testing-library/react';
import Label from '.';
import '@testing-library/jest-dom'

describe('Label', () => {
    it('should render', () => {
        const { container } = render(<Label variant="primary">Test label</Label>);
        expect(container).toMatchSnapshot()
    });

    
    it('should render the children prop', () => {
        const { getByText } = render(<Label variant="primary">Test label</Label>);
        expect(getByText('Test label')).toBeInTheDocument();
    });
});
