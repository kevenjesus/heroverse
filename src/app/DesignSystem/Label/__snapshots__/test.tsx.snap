// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render 1`] = `
<div>
  <div
    class="sc-hLseeU bhjssR"
  >
    <div
      class="sc-eDDNvR fgXnaz"
    />
    <p
      class="sc-dmqHEX hZDiJS"
    >
      Test label
    </p>
  </div>
</div>
`;
